daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/ConverterOps.scala

```diff
@@ -59,7 +59,7 @@ import org.apache.daffodil.xml.XMLUtils
 trait NumericRangeCheck {
   protected val rangePrim: PrimNumeric
   final protected def checkRange(num: Number) = {
-    if (!rangePrim.isValidRange(num)) {
+    if (!rangePrim.isValid(num)) {
       val msg = "Value is out of range for %s type: %s".format(rangePrim.asInstanceOf[PrimType].globalQName, num)
       throw new NumberFormatException(msg)
     }
```

Comment:
I'm wondering if this NumericRangeCheck trait can just go away, and then we change the converters to use the primType fromNumber(), since that now does range checking? For example, I think LongToInteger becomes:
```scala
case object LongToInt extends Converter {
  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueInt = {
    PrimType.Int.fromNumber(a.getLong)
  }
}
```
Consolidating all the conversion and range checking logic into the NodeInfo primitives seems like the right thing to me.